File: src/MyServiceBus/EndpointDefinition.cs
namespace MyServiceBus;

/// <summary>
/// Describes transport-neutral policy for the endpoint that hosts one or more consumers.
/// </summary>
public sealed class EndpointDefinition
{
    private string? name;
    private int? concurrentMessageLimit;
    private ushort? prefetchCount;

    public string? Name
    {
        get => name;
        set
        {
            if (value is not null)
                ArgumentException.ThrowIfNullOrWhiteSpace(value);

            name = value;
        }
    }

    public int? ConcurrentMessageLimit
    {
        get => concurrentMessageLimit;
        set
        {
            if (value is <= 0)
                throw new ArgumentOutOfRangeException(nameof(value), value, "The concurrent message limit must be greater than zero.");

            concurrentMessageLimit = value;
        }
    }

    public ushort? PrefetchCount
    {
        get => prefetchCount;
        set
        {
            if (value is 0)
                throw new ArgumentOutOfRangeException(nameof(value), value, "The prefetch count must be greater than zero.");

            prefetchCount = value;
        }
    }
}

File: src/Java/myservicebus/src/main/java/com/myservicebus/EndpointDefinition.java
package com.myservicebus;

/** Transport-neutral policy for an endpoint that hosts one or more consumers. */
public final class EndpointDefinition {
    private String name;
    private Integer concurrentMessageLimit;
    private Integer prefetchCount;

    public String getName() {
        return name;
    }

    public EndpointDefinition name(String value) {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Endpoint name must not be blank.");
        }
        name = value;
        return this;
    }

    public Integer getConcurrentMessageLimit() {
        return concurrentMessageLimit;
    }

    public EndpointDefinition concurrentMessageLimit(int value) {
        if (value < 1) {
            throw new IllegalArgumentException("Concurrent message limit must be greater than zero.");
        }
        concurrentMessageLimit = value;
        return this;
    }

    public Integer getPrefetchCount() {
        return prefetchCount;
    }

    public EndpointDefinition prefetchCount(int value) {
        if (value < 1) {
            throw new IllegalArgumentException("Prefetch count must be greater than zero.");
        }
        prefetchCount = value;
        return this;
    }
}
